frontend/app/routes/$lang/_protected/applications/$id.tsx

```diff
@@ -117,89 +136,145 @@ export default function ViewApplication() {
-                  <Address
-                    address={applicationDetails.personalInformation?.homeAddress ?? ''}
-                    city={applicationDetails.personalInformation?.homeCity ?? ''}
-                    provinceState={i18n.language === 'en' ? homeProvince?.nameEn : homeProvince?.nameFr}
-                    postalZipCode={applicationDetails.personalInformation?.homePostalCode ?? ''}
-                    country={i18n.language === 'en' ? homeCountry?.nameEn ?? '' : homeCountry?.nameFr ?? ''}
-                    apartment={applicationDetails.personalInformation?.homeApartment}
-                    altFormat={true}
-                  />
+                <DescriptionListItem term={t('applications:view-application.consent')}>
+                  {applicationDetails.partnerInformation.confirm ? t('applications:view-application.consent-answer-yes') : t('applications:view-application.consent-answer-no')}
                 </DescriptionListItem>
               </dl>
             </section>
+          )}
+          {applicationDetails.children && (
             <section className="space-y-6">
-              <section className="space-y-6">
-                <h2 className="font-lato text-2xl font-bold">{t('applications:view-application.comm-prefs')}</h2>
-                <dl className="divide-y border-y">
-                  <DescriptionListItem term={t('applications:view-application.comm-pref')}>
-                    <p>{communicationPreferenceLang}</p>
-                    {applicationDetails.communicationPreferences?.email && <p>{applicationDetails.communicationPreferences.email}</p>}
-                  </DescriptionListItem>
-                  <DescriptionListItem term={t('applications:view-application.lang-pref')}>{preferredLanguage}</DescriptionListItem>
-                </dl>
-              </section>
+              <h2 className="font-lato text-2xl font-bold">{t('applications:view-application.child-information.page-title')}</h2>
+              {applicationDetails.children.map((child, key) => {
+                const childName = `${child.information.firstName} ${child.information.lastName}`;
+                const childDateOfBirth = child.information.dateOfBirth ? toLocaleDateString(parseDateTimeString(child.information.dateOfBirth), i18n.language) : '';
+                const childFederalSocialProgram = allFederalSocialPrograms
+                  .filter((federalProgram) => federalProgram.id === child.dentalBenefits?.federalSocialProgram)
+                  .map((obj) => getNameByLanguage(locale, obj))
+                  .join(', ');
+                const childProvincialTerritorialSocialProgram = allProvincialTerritorialSocialPrograms
+                  .filter((provinceTerritorialProgram) => provinceTerritorialProgram.id === child.dentalBenefits?.provincialTerritorialSocialProgram)
+                  .map((obj) => getNameByLanguage(locale, obj))
+                  .join(', ');
```

Comment:
Same comment as the other one I made about `federalSocialProgram` and `provincialTerritorialSocialProgram`, which can be applied to the child.

Reply:
I see what you mean and dislike the idea of maintaining a separate array. We should consider having the service layer be responsible for mapping the ID into an object with `nameEn`/`nameFr` to avoid this. 

I will resolve but we should add a `TODO` in the `loader(..)` where `allFederalSocialPrograms` and `allProvincialTerritorialSocialPrograms` is returned to indicate that it's not necessary to return all programs to the component.